integrationtests/self/http_test.go

```diff
@@ -1068,4 +1068,105 @@ var _ = Describe("HTTP tests", func() {
 			"Unannounced": {"Surprise!"},
 		})))
 	})
+
+	It("graceful shutdown successfully with no timeout", func() {
+		var (
+			startShutdownChan   = make(chan struct{})
+			shutdownStartedChan = make(chan struct{})
+			clientDoneChan      = make(chan struct{})
+			serverDoneChan      = make(chan struct{})
+		)
+
+		mux.HandleFunc("/slow", func(w http.ResponseWriter, r *http.Request) {
+			close(startShutdownChan)
+			<-shutdownStartedChan
```

Comment:
This channel doesn't do anything, as it's immediately closed in the `CloseGracefully` Go routine.

What you should do instead is use a `time.Sleep(250*time.Millisecond)` here. That will (in practice) guarantee that the `Write` is run after `CloseGracefully`.